packages/app/src/resource/AppsPage.tsx

```diff
@@ -24,18 +31,57 @@ export function AppsPage(props: AppsPageProps): JSX.Element {
     );
   }
 
+  function launchApp(clientApplication: ClientApplication): void {
+    const smartAppLaunch: SmartAppLaunch = {
+      resourceType: 'SmartAppLaunch',
+    };
+
+    switch (props.resource.resourceType) {
+      case 'Patient':
+        smartAppLaunch.patient = createReference(props.resource);
+        break;
+      case 'Encounter':
+        smartAppLaunch.patient = props.resource.subject as Reference<Patient>;
+        smartAppLaunch.encounter = createReference(props.resource);
+        break;
+    }
+
+    medplum
+      .createResource(smartAppLaunch)
+      .then((result) => {
+        const url = new URL(clientApplication.launchUri as string);
+        url.searchParams.set('iss', medplum.getBaseUrl() + 'fhir/R4');
+        url.searchParams.set('launch', result.id as string);
+        window.location.assign(url.toString());
+      })
+      .catch((err) => showNotification({ color: 'red', message: normalizeErrorString(err) }));
+  }
+
   return (
     <Document>
       {questionnaires.map((questionnaire) => (
         <div key={questionnaire.id}>
-          <h3>
+          <Title order={3}>
```

Comment:
Unrelated to this PR, but it would be better for this link to read from `questionnaire.title` rather than `questionnaire.name`. 

`questionnaire.name` is supposed to be a more computer-friendly identifier, but the `title` is the human readable one

Reply:
I updated it to prefer `questionnaire.title`, but still fallback to `questionnaire.name`, because it looks like there are a bunch of questionnaires with names but not titles.